manager/src/main/java/io/gingersnapproject/RuleResource.java

```diff
@@ -23,19 +25,29 @@ public class RuleResource {
    @Inject
    QueryHandler queryHandler;
 
+   @Inject
+   RuleManager ruleManager;
+
    @GET
    @Operation(summary = "Retrieve a cache entry associated with the provided rule and key")
    @Path("/{rule}/{key}")
    @Produces(MediaType.APPLICATION_JSON)
    public Uni<String> get(String rule, String key) {
-      return maps.get(rule, key);
+      checkRuleExists(rule);
+      return maps.get(rule, key)
+              .onItem()
+              .invoke(v -> {
+                 if (v == null || v.isEmpty())
+                    throw new NotFoundException();
+              });
```

Comment:
This should allow the happy path to have no additional allocations on our part.
```suggestion
      checkRuleExists(rule);
      Uni<String> uni = maps.get(rule, key);
      if (uni instanceof UniItem uniItem)  {
         if (unitItem.getItem() == null) {
            throw new NotFoundException();
         }
         return uni;
      }
      return uni.onItem()
         .invoke(v -> {
                 if (v == null || v.isEmpty())
                    throw new NotFoundException();
              });
```

Reply:
Also I wonder should empty String be a 404? That sounds like a 204 in that case no? Either way it needs to be consistent in the code above, I have it inconsistent right now.

Reply:
> That sounds like a 204 in that case no?

If we're going to support empty values, then yes.

Luckily Quarkus handles this case for us, so I have removed this check.